src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -762,55 +766,72 @@ private File resolveArtifactSources( Artifact artifact )
 
     private List<ResourceTransformer> getResourceTransformers()
     {
+        final List<ResourceTransformer> resourceTransformers = new LinkedList<>( getDefaultResourceTransformers() );
         if ( transformers == null )
         {
-            return Collections.emptyList();
+            return resourceTransformers;
```

Comment:
You're still merging. Here you should replace `Collections.emptyList()` with `getDefaultResourceTransformers()`

Reply:
this is the case, resourceTransformers is initialized with getDefaultResourceTransformers. If the else skips the default transformers the feature is pointless since in 80% of the case you need to add a custom transformer or customize a config (like mainClass for the manifest)

Reply:
Let's agree to disagree

Reply:
I'm fine to disagree but I'd like to solve the issue which aims to reduce the entry cost of the plugin and save hours of debug time due to a bad packaging which is hard to identify - and even more with jpms since module-info.java drops SPI registration :s.

any alternative proposal to have a trivial to enable good auto configuration compatible with user customizations?

Reply:
Can't think of a clean, clear and simple solution yet. So let's split it into 2 issues: 1. have good default transformers; 2. combine default+custom transfomers

Reply:
Ok, let's split. Just to ensure I get it and not do work for nothing, it means 3 PR - no order:

1. defaults (transformers and filters)
2. SPI
3. combine
?

Reply:
1 done at https://github.com/apache/maven-shade-plugin/pull/12